### PR TITLE
Api test gardening

### PR DESCRIPTION
**Pull request by ChrisHines**
## What does this PR (Pull Request) do?
- [x] This PR makes a few small improvements to the traffic_ops API tests toward more idiomatic use of Go. It is not related to any issue.

## Which Traffic Control components are affected by this PR?
- Traffic Ops

## What is the best way to verify this PR?
Review the changes and run the tests.

## The following criteria are ALL met by this PR
- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

## Additional Information
The commit messages for each commit in this PR describe the class of code improvement made in that commit.

### CODE REVIEW

File: traffic_ops/testing/api/v14/cdnfederations_test.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func UpdateTestCDNFederations(t *testing.T) {
 		log.Debugf("GET Response: %s\n", bytes)
 
 		if resp2.Response[0].CName == nil {
-			log.Errorf("CName is nil after updating\n")
+			log.Errorln("CName is nil after updating")
```
Comment (mhoppa) [inline traffic_ops/testing/api/v14/cdnfederations_test.go]: should this just be `log.Error` as you did in other spots?

File: traffic_ops/testing/api/v14/profiles_test.go [hunk 2]
```diff
@@ -69,33 +69,33 @@ func CreateTestProfiles(t *testing.T) {
 			t.Errorf("could not GET profile with name: %s %v\n", pr.Name, err)
 		}
 		if len(profiles) == 0 {
-			t.Errorf("could not GET profile %++v: not found\n", pr)
+			t.Errorf("could not GET profile %+v: not found\n", pr)
```
Comment (mhoppa) [inline traffic_ops/testing/api/v14/profiles_test.go]: does ErrorF need `\n` I dont think it does

File: traffic_ops/testing/api/v14/cdnfederations_test.go (hunk 1 above)
Comment (ChrisHines) [reply traffic_ops/testing/api/v14/cdnfederations_test.go]: There is no `log.Error` function in `lib/go-log`. See https://godoc.org/github.com/apache/trafficcontrol/lib/go-log.

I believe nearly all the other similar changes this PR makes are calling `t.Error` where `t` is a `*testing.T`.

File: traffic_ops/testing/api/v14/profiles_test.go (hunk 2 above)
Comment (ChrisHines) [reply traffic_ops/testing/api/v14/profiles_test.go]: Nope, not sure how I missed this one. Thanks, I will update it.

Comment (ChrisHines) [reply traffic_ops/testing/api/v14/profiles_test.go]: Oh, I see. My regex was only looking for things ending in `\n")` and that closing paren blocked finding this form.

File: traffic_ops/testing/api/v14/cdnfederations_test.go (hunk 1 above)
Comment (mhoppa) [reply traffic_ops/testing/api/v14/cdnfederations_test.go]: ahh shoot I didnt even see this was a log instead of t.Error my bad :)

Comment (ChrisHines) [reply traffic_ops/testing/api/v14/cdnfederations_test.go]: No worries, I did the same thing while working on this PR.

File: traffic_ops/testing/api/v14/federations_test.go [hunk 3]
```diff
@@ -16,6 +16,7 @@ package v14
 */
 
 import (
+	"errors"
```
Comment (ocket8888) [inline traffic_ops/testing/api/v14/federations_test.go]: Unused import?

Comment (ChrisHines) [reply traffic_ops/testing/api/v14/federations_test.go]: https://github.com/apache/trafficcontrol/pull/4134/files/d036d830cf1e408ecc8054f9e63964d586c8718a#diff-e8648860be2706c170d6d9ac748d2f24L90

Comment (ocket8888) [reply traffic_ops/testing/api/v14/federations_test.go]: oh I see. That link is broken for some reason btw - doesn't take me anywhere specific.

Comment (ChrisHines) [reply traffic_ops/testing/api/v14/federations_test.go]: The link should work, but it does have a noticeable delay jumping to the specific line while the full page loads.

Comment (ocket8888) [reply traffic_ops/testing/api/v14/federations_test.go]: I think it's because I have all the files marked as "viewed" and it's not opening them

**Pull request merged**